### DISCUSSION TITLE
Download reports as PDF

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#1652] [Anything Else]
"I've had one question from a client about PDF reports. Is this possible, or on the horizon with Plausible? I've advised them to do a whole-page screenshot for now, or print and then save as a PDF. Their issue though was that they want the Top Pages, Entry Pages and Exit Pages to all show, rather than being hidden by tabs."

(request via email)

Our [CSV export](https://plausible.io/docs/export-stats) is a good alternative to PDF reports for now

**Comment by Vigasaurus** [comment]
I feel like a really nice implementation for something like this would be to allow creating shared links that have a filter pre-applied that cannot be removed - this would allow for you to limit what range of data/which sources etc can be seen from a shared link, whilst still providing someone with the full plausible detail and functionality within that subset of data.

I could even see this being useful for sharing an affiliate's dashboard based on UTM tags etc. 

Even better would be the ability to save the links with a name/short link directly, so you could share something like https://plausible.io/example.com/share/AffiliateName

And potentially the ability to hide specific reports or similar in a such shared link (or any shared link, for that matter).

**Reply by metmarkosaric** [reply]
nice idea @Vigasaurus! i can see that being very useful!

**Reply by Vigasaurus** [reply]
Oh this suggestion already exists in #1495 😅

**Reply by ukutaht** [reply]
Yes this would be a very cool addition

**Comment by ghenry** [comment]
I was just looking at Elixir PDF libs for this as I wanted to email myself PDF stats reports. I just did (out in 2 months though at https://www.se-radio.net/team/gavin-henry/ ) a podcast episide for [SE Radio](https://www.se-radio.net/) about 30 Years of PDF and when chatted about always sending a PDF version of HTML email reports to users to preserve presentation etc.

Any work done on this or planned?

**Comment by ghenry** [comment]
This looks OK. I installed the tracker on Thursday night last week and enabled the weekly report email, which arrived this morning.

I've cleaned up the HTML from the email source to make sure it looks the same (attached) and then, using [ChromicPDF](https://hexdocs.pm/chromic_pdf/readme.html) added it to my test Phoenix app and run in iex:

```
iex(4)> ChromicPDF.print_to_pdf({:url, "https://www.surevoip.co.uk/plausible.html"}, output: "surevoip_plausible_report.pdf")
:ok
```
which produces attached. Not bad looking.

[surevoip_plausible_report.pdf](https://github.com/plausible/analytics/files/9339457/surevoip_plausible_report.pdf)
[plausible.html.txt](https://github.com/plausible/analytics/files/9339472/plausible.html.txt)

it does mean headless chrome (https://github.com/bitcrowd/chromic_pdf/blob/main/lib/chromic_pdf/pdf/chrome_impl.ex#L43) is needed though (ChromicPDF is a HTML-to-PDF renderer for Elixir, based on headless Chrome), which is quite a requirement. Thoughts?

**Reply by ghenry** [reply]
@metmarkosaric Any thoughts?
